1 Quick Sort/QuickSort1.py

```diff
@@ -0,0 +1,89 @@
+import random
+import time
+import os
+numbersArr = []
+path = input("Inter path to the file:")
+
+def readArrayFromFile(numbersArr,path):
+        file = open(path +".py")
+        text = ""
+        for line in file:
+            text = text + line
+        numbersArr = toIntArray(text.split())
+        print(len(numbersArr))
+        return numbersArr
+
+def makeSortedArr(arr, path):
+
+    fileName = os.path.basename(path)
+    reversedFileName = reverseFileName(fileName)
+    file = open((fileName+".py"),"tw")
+    file.write(str(sortedArr))
+    file.close()
+        fileName = os.path.basename(path)
+        if arr == ("reversedSortedArr "):
+                reversedFileName = reverseFileName(fileName)
+                open((reversedFileName+".py"),"tw")
+                file.write(str(arr))
+                file.close()
+        elif arr == ("sortedArr"):
+                filename = "Sorted "+filename+".py"
+                file = open((filename),"tw")
+                file.write(str(arr))
+                file.close()
+
+def reverseFileName(string):
+        invertedString = ""
+        length = len(string)
+        for index in range(0, length):
+                invertedString = invertedString + string[length-index-1]
+        return invertedString
+
+def toIntArray(arr):
+    intArr = []
+    for symbol in arr:
+        digit = int(symbol)
+        intArr.append(digit)
+    return  intArr
+
+
+def quicksort(arr, direction):
+    if len(arr) <= 1:
+        return arr
+    else:
+        randomNum = random.choice(arr)
+        sArr = []
+        mArr = []
+        eArr = []
+        if direction == "+":
```

Comment:
Code under ```direction == "+"``` and ```direction == "-"``` is pretty much the same.
The basic programming principle DRY (don't repeat yourself) said that all the common code shouldn't be copy-pasted and updated. In such case the best idea is to move bunch of code to the separate function:
```
for element in arr:
                if element < <...>: 
                    <...>.append(element)
                elif element > randomNum:
                    <...>.append(element)
                else:
                    <...>.append(element)
            return quicksort(<...>,<...>) + <...> + quicksort(<...>,<...>)
```
where <...> is function input values (they will be different in direction == "+" and direction == "-" calls)